### PR TITLE
chore(all): Upgrade GQL and dependencies

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* We want to use the latest graphql dependency

This commit:
* Updates GQL dep and all packages relying on it
* Moves some dependencies to root package.json
* Removes a couple of deps we aren't using
* Addresses breaking changes, updates related tests and mocks
* Fixes type errors
* Prettifies contentful autogenerated files

closes FXA-8489

---

New branch in favor of #15757 for a cleaner merge with main.

### CODE REVIEW

File: libs/shared/contentful/src/__generated__/index.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
-export * from "./fragment-masking";
-export * from "./gql";
\ No newline at end of file
+export * from './fragment-masking';
+export * from './gql';
```
Comment (julianpoy) [inline libs/shared/contentful/src/__generated__/index.ts:2]: Interesting, prettier differences?

File: packages/fxa-admin-panel/package.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
     ]
   },
   "dependencies": {
-    "@apollo/client": "^3.4.5",
+    "@apollo/client": "^3.8.3",
```
Comment (bbangert) [inline packages/fxa-admin-panel/package.json:39]: We can root this dep on a follow-up.

File: packages/fxa-event-broker/package.json [hunk 3]
```diff
@@ -51,7 +51,7 @@
     "fxa-jwtool": "^0.7.2",
     "fxa-shared": "workspace:*",
     "google-auth-library": "^9.2.0",
-    "graphql": "^15.6.1",
+    "graphql": "^16.8.0",
```
[the remark this answers is not in the document]
Comment (julianpoy) [reply packages/fxa-event-broker/package.json:54]: As well as all of the other updated dependencies, unless they conflict in some way and cannot be moved.

**Comment by LZoog** [comment]
For posterity, I'd left a few comments on the other branch:
* https://github.com/mozilla/fxa/pull/15757#discussion_r1410858575
* https://github.com/mozilla/fxa/pull/15757#discussion_r1410864264
* https://github.com/mozilla/fxa/pull/15757#discussion_r1410865901

**Review by bbangert — APPROVED** [review]

**Review by julianpoy — APPROVED** [review]
lgtm after dependencies are moved to root package.json, but also okay if that was supposed to be a follow-up ticket

File: yarn.lock [hunk 4]
```diff
@@ -53639,8 +52400,8 @@ fsevents@~2.1.1:
   linkType: hard
 
 "protobufjs@npm:6.11.3, protobufjs@npm:^6.10.0, protobufjs@npm:^6.11.2, protobufjs@npm:^6.11.3, protobufjs@npm:^6.8.8":
-  version: 6.11.4
-  resolution: "protobufjs@npm:6.11.4"
+  version: 6.11.3
```
Comment (dschom) [inline yarn.lock:52403]: dinged by dependabot?

**Pull request merged**